Lib/test/test_asyncio/test_eager_task_factory.py

```diff
@@ -0,0 +1,241 @@
+"""Tests for base_events.py"""
+
+import gc
+import time
+import unittest
+from types import GenericAlias
+from unittest import mock
+
+import asyncio
+from asyncio import base_events
+from asyncio import tasks
+from test.test_asyncio import utils as test_utils
+from test.test_asyncio.test_tasks import get_innermost_context
+from test import support
+
+MOCK_ANY = mock.ANY
+
+
+def tearDownModule():
+    asyncio.set_event_loop_policy(None)
+
+
+class EagerTaskFactoryLoopTests(test_utils.TestCase):
+
+    def new_task(self, loop, coro, name='TestTask', context=None):
+        return tasks.Task(coro, loop=loop, name=name, context=context)
+
+    def new_future(self, loop):
+        return asyncio.Future(loop=loop)
+
+    def setUp(self):
+        super().setUp()
+        self.loop = asyncio.new_event_loop()
+        self.loop.set_task_factory(asyncio.eager_task_factory)
+        self.set_event_loop(self.loop)
+
+    def test_eager_task_factory_set(self):
+        self.assertIs(self.loop.get_task_factory(), asyncio.eager_task_factory)
+
+    def test_await_future_during_eager_step(self):
+
+        async def set_result(fut, val):
+            fut.set_result(val)
+
+        async def run():
+            fut = self.loop.create_future()
+            t = self.loop.create_task(set_result(fut, 'my message'))
+            # assert the eager step completed the task
+            self.assertTrue(t.done())
+            return await fut
+
+        result = self.loop.run_until_complete(run())
+        self.assertEqual(result,'my message')
+
+    def test_eager_completion(self):
+
+        async def coro():
+            return 'hello'
+
+        async def run():
+            await asyncio.sleep(0.1)
```

Comment:
I'm not sure this adds anything. What's your intent with this sleep (and the others also in tests' `run()` functions)?

Reply:
in some cases I needed this to make sure the event loop started before doing anything that expects to take an eager step, because the eager step is taken *only* if the event loop is already running.

I refactored the test code to make this clearer, and removed the sleep where it wasn't actually needed.